script/create-test-appx.js

```diff
@@ -0,0 +1,224 @@
+      if (code !== 0) {
+        return reject(new Error(fileName + ' exited with code: ' + code))
+      }
+      return resolve(data.join('\n'))
+    })
+
+    child.stdin.end()
+  })
+}
+
+/**
+ * Installs the test certificate to the local machine.
+ * Needs admin rights.
+ */
+async function installPfx () {
+  const installPfxArgs = [
+    'Import-PfxCertificate',
+    '-FilePath',
+    CERT_PATH,
+    '-CertStoreLocation',
+    '"Cert:\\LocalMachine\\TrustedPeople"'
+  ]
+
+  await executeChildProcess('powershell.exe', installPfxArgs)
+}
+
+/**
+ * Is the PFX certificate installed on the current machine?
+ *
+ * @returns {boolean}
+ */
+async function isPfxInstalled () {
+  const checkPfxArgs = [
+    '[bool](dir cert:\\LocalMachine\\TrustedPeople\\ | ? { $_.subject -like "cn=electrontest" })'
+  ]
+
+  const result = await executeChildProcess('powershell.exe', checkPfxArgs)
+
+  return result.toLowerCase().trim() === 'true'
+}
+
+/**
+ * Copy in our known shim files
+ *
+ * @returns {Promise<void>}
+ */
+async function copyInShim () {
+  const contents = await fs.readdir(SHIM_PATH)
+
+  for (const file of contents) {
+    const source = path.join(SHIM_PATH, file)
+    const desination = path.join(PREAPPX_PATH, file)
+
+    await fs.copy(source, desination)
+  }
+}
+
+/**
+ * Take the output directory and turn it into an appx
+ *
+ * @returns {Promise<void>}
+ */
+async function convert () {
+  fs.emptyDirSync(OUT_PATH)
+
+  // Install the test certificate
+  if (!(await isPfxInstalled())) {
+    console.warn(`The test certificate is not installed!`)
+    console.warn(`We will now try to install it. If this is not an administrative shell,`)
+    console.warn(`this operation will fail. If that's the case, please run this script`)
+    console.warn(`again as administrator`)
+
+    await installPfx()
+  }
+
+  await electronWindowsStore({
+    inputDirectory: STAGING_PATH,
+    outputDirectory: OUT_PATH,
+    packageVersion: '1.0.0.0',
+    packageName: 'electron',
+    manifest: MANIFEST_PATH,
+    assets: ASSETS_PATH,
+    devCert: CERT_PATH,
+    createPriParams: ['/IndexName', '99999D7F.Electron'],
+    makePri: true,
+    finalSay: copyInShim
+  })
+}
+
+/**
+ * Remove staging directories
+ *
+ * @returns {Promise<void>}
+ */
+async function cleanup () {
+  await fs.remove(STAGING_PATH)
+}
+
+/**
+ * Main script method: Create an appx
+ *
+ * @returns {Promise<void>}
+ */
+async function main () {
+  if (!fs.existsSync(BUILD_PATH)) {
+    console.log(`Cannot find ${BUILD_PATH}`)
+    return
+  }
+
+  fs.emptyDirSync(STAGING_PATH)
+
+  const dirContents = fs.readdirSync(BUILD_PATH)
+
+  // Copy things over
+  for (const item of dirContents) {
+    const source = path.join(BUILD_PATH, item)
+    const destination = path.join(STAGING_PATH, item)
+    const itemStat = fs.statSync(source)
+    const isNeededFile = itemStat.isFile() && isFileNeeded(item)
+    const isNeededDir = !isNeededFile && isDirNeeded(item)
```

Comment:
By checking for `!isNeededFile` instead of `itemStat.isDirectory()` here, is it possible for `isNeededDir` to be true when `item` is a file rather than a directory?

Reply:
I just wanted to avoid calling the method, but the difference in performance is so small that we should just keep it easy. I'll change it to `itemStat.isDirectory()`!